pkg/server/downloader.go

```diff
@@ -106,6 +109,31 @@ func (d *BackingImageDownloader) Send(filePath string, address string) error {
 	return d.DownloaderEngine.SenderLaunch(filePath, address, types.FileSyncTimeout, false)
 }
 
+func (d *BackingImageDownloader) Upload(port string, directory string, syncFileOps sparserest.SyncFileOperations, sizeUpdater uploadserver.SizeUpdater) error {
```

Comment:
Not sure if there is a better design for this API. Right now inputs `syncFileOps sparserest.SyncFileOperations` and `sizeUpdater uploadserver.SizeUpdater` are actually the same BackingImage object